### DISCUSSION TITLE
Osm2pgsql 1.6 data dump to specific schema of DB

### DISCUSSION THREAD

**Discussion by DeepanshuRai-TomTom** [osm2pgsql-dev/osm2pgsql#1686] [Q&A] [answered]
While using osm2pgsql 1.6 i am trying to dump PBF into specific schema of postgres DB using following commands using docker. (i am setting database properties in docker env)

docker run -v /mnt/ne-blobs/:/mnt/ne-blobs/ --env PGHOST=$PGHOST --env PGPORT=$PGPORT --env PGUSER=$PGUSER --env PGDATABASE=$PGDATABASE --env PGPASSWORD=$PGPASSWORD osm2pgsql:1.6.0 **osm2pgsql -c -E 4326 --hstore --slim -C 8000 --output-pgsql-schema=$schema_name -r pbf /path/to/pbf.**

I am creating schema before executing this command but i see that only 
planet_osm_line, planet_osm_point, planet_osm_polygon, planet_osm_roads

are getting created in my mentioned schema

but core tables like Planet_osm_nodes, planet_osm_rels ,planet_osm_ways

are getting created in public schema. Is there anything i am doing wrong with the command? i want to create all 7 tables in one schema which is execution specific.

what is the role of --middle-schema ??

**Comment by joto** [comment, marked as answer]
`--middle-schema` is exactly the command line option you are looking for.

**Reply by DeepanshuRai-TomTom** [reply]
--middle-schema along with --output-pgsql-schema or just --middle-schema?
